lib/flipper/test/shared_adapter_test.rb

```diff
@@ -0,0 +1,239 @@
+    @time_gate =  @feature.gate(:percentage_of_time)
+    Flipper.register(:admins) do |actor|
+      actor.respond_to?(:admin?) && actor.admin?
+    end
+
+    Flipper.register(:early_access) { |actor|
+      actor.respond_to?(:early_access?) && actor.early_access?
+    }
+  end
+
+  def teardown
+    super
+    Flipper.unregister_groups
+  end
+
+  def test_name_is_a_symbol
+    refute_empty  @adapter.name
+    assert_kind_of Symbol, @adapter.name
+  end
+
+  def test_has_included_the_flipper_adapter_module
+    assert_includes  @adapter.class.ancestors, Flipper::Adapter
+  end
+
+  def test_returns_correct_default_values_for_gates_if_none_are_enabled
+    expected = {
+      :boolean => nil,
+      :groups => Set.new,
+      :actors => Set.new,
+      :percentage_of_actors => nil,
+      :percentage_of_time => nil,
+    }
+    assert_equal expected, @adapter.get(@feature)
+  end
+
+  def test_can_enable_and_get_value_for_boolean_gate
+    assert_equal true, @adapter.enable(@feature, @boolean_gate, @flipper.boolean)
+    assert_equal 'true', @adapter.get(@feature)[:boolean]
+  end
+
+  def test_can_disable_and_get_value_for_boolean_gate
+    assert_equal true, @adapter.disable(@feature, @boolean_gate, @flipper.boolean(false))
+    assert_equal nil, @adapter.get(@feature)[:boolean]
+  end
+
+  def test_disables_all_enabled_things_when_boolean_gate_disabled
+    actor_22 = @actor_class.new('22')
+    assert_equal true, @adapter.enable(@feature, @boolean_gate, @flipper.boolean)
+    assert_equal true, @adapter.enable(@feature, @boolean_gate, @flipper.group(:admins))
+    assert_equal true, @adapter.enable(@feature, @boolean_gate, @flipper.actor(actor_22))
+    assert_equal true, @adapter.enable(@feature, @boolean_gate, @flipper.actors(25))
+    assert_equal true, @adapter.enable(@feature, @boolean_gate, @flipper.time(45))
+    assert_equal true, @adapter.disable(@feature, @boolean_gate, @flipper.boolean(false))
+    expected = {
+      :boolean => nil,
+      :groups => Set.new,
+      :actors => Set.new,
+      :percentage_of_actors => nil,
+      :percentage_of_time => nil,
+    }
+    assert_equal expected, @adapter.get(@feature)
+  end
+
+  def test_can_enable_disable_get_value_for_group_gate
+    assert_equal true, @adapter.enable(@feature, @group_gate, @flipper.group(:admins))
+    assert_equal true, @adapter.enable(@feature, @group_gate, @flipper.group(:early_access))
+
+    result = @adapter.get(@feature)
+    assert_equal Set['admins', 'early_access'], result[:groups]
+
+    assert_equal true, @adapter.disable(@feature, @group_gate, @flipper.group(:early_access))
+    result = @adapter.get(@feature)
+    assert_equal Set['admins'], result[:groups]
+
+    assert_equal true, @adapter.disable(@feature, @group_gate, @flipper.group(:admins))
+    result = @adapter.get(@feature)
+    assert_equal Set.new, result[:groups]
+  end
+
+  def test_can_enable_disable_and_get_value_for_an_actor_gate
```

Comment:
great catch sorry for missing that.  Will double check that I've covered all of the tests